codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -510,31 +499,15 @@ public void onUnknownFrame(ChannelHandlerContext ctx, byte frameType, int stream
             onUnknownFrame0(ctx, frameType, streamId, flags, payload);
         }
 
-        /**
-         * Indicates whether or not frames for the given stream should be ignored based on the state of the
-         * stream/connection.
-         */
-        private boolean shouldIgnoreFrame(Http2Stream stream, boolean allowResetSent) {
-            if (connection.goAwaySent() &&
-                    (stream == null || connection.remote().lastStreamCreated() <= stream.id())) {
-                // Frames from streams created after we sent a go-away should be ignored.
-                // Frames for the connection stream ID (i.e. 0) will always be allowed.
-                return true;
-            }
-
-            // Also ignore inbound frames after we sent a RST_STREAM frame.
-            return stream != null && !allowResetSent && stream.isResetSent();
+        private boolean streamCreatedAfterGoAwaySent(Http2Stream stream) {
+            int lastStreamId = connection().remote().lastKnownStream();
+            // Ignore inbound frames after a GOAWAY was sent and either the stream does not exist (anymore) or
+            // the streams id is greater than the last stream id set in the GOAWAY frame.
+            return connection.goAwaySent() && (stream == null || stream.id() > lastStreamId);
         }
 
-        /**
-         * Verifies that a GO_AWAY frame was not previously received from the remote endpoint. If it was, throws a
-         * connection error.
-         */
-        private void verifyGoAwayNotReceived() throws Http2Exception {
-            if (connection.goAwayReceived()) {
-                // Connection error.
-                throw connectionError(PROTOCOL_ERROR, "Received frames after receiving GO_AWAY");
-            }
+        private boolean resetSentForStream(Http2Stream stream) {
+            return stream != null && stream.isResetSent();
```

Comment:
@nmittler hmm. I think we don't need the method at all. We really should just check if the stream is closed and throw away data, headers and window update frames for closed streams.

Regarding the stream history. Wouldn't it be enough instead of the `CloseState` to have a flag `resetSent`. This is really the only case when we would like to offer some kind of "grace period" when we still accept frames. WDYT?